### PR TITLE
Split Config class to dedicated feature config

### PR DESCRIPTION
**Pull request by PerfectSlayer**
# What Does This Do

This PR splits the `Config` class to dedicated feature config classes.

# Motivation

Having multiple configuration class will help to have different initialization stages for GraalVM `native-image`.
It should also help to maintain configuration and have a clearer idea of how is defined config values and where they are used.

# Additional Notes

The `Config` API was maintained by delegating calls to the new config classes.

### CODE REVIEW

File: internal-api/src/main/java/datadog/trace/api/config/TraceInstrumentationFeatureConfig.java [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+  private final boolean logsMDCTagsInjectionEnabled;
+  private final String traceMethods;
+  private final boolean traceExecutorsAll;
+  private final List<String> traceExecutors;
+  private final Set<String> traceThreadPoolExecutorsExclude;
+  private final List<String> excludedClasses;
+  private final String excludedClassesFile;
+  private final Set<String> excludedClassLoaders;
+  private final List<String> excludedCodeSources;
+  private final boolean httpServerTagQueryString;
```
Comment (mcculls) [inline internal-api/src/main/java/datadog/trace/api/config/TraceInstrumentationFeatureConfig.java:97]: Note this is still mixing build-time instrumentation config (what classes to exclude, custom methods to trace, etc.) with run-time config that's used by the injected advice and/or decorators (such as HTTP tag settings)

We still want to load the advice config at run-time so people can change the content of trace data.

Could we maybe start with a smaller PR that just separates out config that's referenced when configuring the byte-buddy agent, but doesn't include config that's accessed from advice classes or their helpers/decorators?

Comment (PerfectSlayer) [reply internal-api/src/main/java/datadog/trace/api/config/TraceInstrumentationFeatureConfig.java:97]: Yes, it is still a draft. Until now, I only apply the split that was already declared in the `dd-trace-api/src/main/java/datadog/trace/api/config/XXXConfig.java` classes.

Then, I wanted to extract from each new `internal-api/src/main/java/datadog/trace/api/config/XXXFeatureConfig.java` what is build-time related. It is not done yet but it must be easier now that the whole `Config` is splitted.

I also still have some static method in the `Config` class that needs to be moved to the feature config they belong to.

**Comment by ValentinZakharov** [comment]
Will it be implemented selective loading based on enabled features/products (appsec, debugger, profiling, etc..) ?

**Review by mcculls — CHANGES_REQUESTED** [review]
I believe this can be closed as stale - it's still a good idea to split up `Config` into smaller pieces, but that would be best done in a new PR that takes the latest updates into account

**Pull request closed without merging**